### ISSUE TITLE
Rebinder rebinds on beans +produced+ by the BeanFactory

### ISSUE THREAD

**Issue by brenuart** [spring-cloud/spring-cloud-commons#187]
Suppose the following configuration:
```
@Configuration
public class RebindConfiguration {
	@Bean
	@ConfigurationProperties(prefix="pojo")
	public PojoFactoryBean pojoFactory() {
		return new PojoFactoryBean();
	}
	
	public static class PojoFactoryBean implements FactoryBean<Pojo> {
		private String name;
		
		public void setName(String name) {
			this.name = "factory-"+name;
		}

		@Override
		public Pojo getObject() throws Exception {
			Pojo p = new Pojo();
			p.name = name;
			return p;
		}
                ...
	}
	
	public static class Pojo {
		private String name;
		
		public void setName(String name) {
			this.name = "bean-"+name;
		}
	}
}
```

The `@ConfigurationProperties` annotation is set on the _FactoryBean_. However, the `ConfigurationPropertiesRebinder` will attempt to rebind on the created *Bean* instead of the factory.

I understand it makes little sense to rebind the properties on the factory after it has produced its instance (unless not a singleton maybe). However, I wasn't expecting the properties to be injected in the bean itself since it isn't annotated. 

Is this an expected behaviour?
How can I avoid it?

**Comment by spencergibb** [comment]
So the `Pojo` is getting recreated?

**Comment by brenuart** [comment]
Here is the scenario:
- the factory is created, and properties injected
- `factory.getObject()` is invoked and the pojo is created
- later, the `ConfigurationPropertiesRebinder` kicks in are re-injected properties on the *Pojo* but not on the factory.

The rebinder rebinds properties on the pojo twice actually. The first time _after singleton instantiation_, the second time... I don't know why.

**Comment by spencergibb** [comment]
That is odd. Is the workaround is to move the `@ConfigurationProperties` out of the factory? Does that help?

**Comment by brenuart** [comment]
Having `@ConfigurationProperties` on the factory method causes the problem (the example above).
Setting the annotation on the factory class instead (`PojoFactoryBean` in the example) doesn't cause the problem: properties are injected on the factory and the rebinder doesn't rebind on the created pojo.

AFAIK the rebinder looks up beans by name in the context when it rebinds properties. I'm wondering if this could be the problem... Is it correct the factory in the above example is named `&pojoFactory` whereas the pojo is named `pojoFactory`? Could this confuse the rebinder?

**Comment by dsyer** [comment]
I don't see how there could be a difference (putting the annotation on the factory bean class declaration doesn't change the bean names).

**Comment by dsyer** [comment]
Fixed in 70d601df84
